client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx

```diff
@@ -73,6 +100,19 @@ CustomDomainPurchaseDetail.propTypes = {
 	onlyBlogDomain: PropTypes.bool,
 	selectedSite: PropTypes.oneOfType( [ PropTypes.bool, PropTypes.object ] ).isRequired,
 	hasDomainCredit: PropTypes.bool,
+	hasNonPrimaryDomainsFlag: PropTypes.bool,
+	siteDomains: PropTypes.array,
 };
 
-export default localize( CustomDomainPurchaseDetail );
+export default connect( state => {
+	const siteId = getSelectedSiteId( state );
+	const siteDomains = getDomainsBySiteId( state, siteId );
+	const registeredDomains = getRegisteredDomains( siteDomains );
+
+	return {
+		hasNonPrimaryDomainsFlag: getCurrentUser( state )
+			? currentUserHasFlag( state, NON_PRIMARY_DOMAINS_TO_FREE_USERS )
+			: false,
+		registeredDomain: head( registeredDomains ),
```

Comment:
This assumption that the first item in this list feels like it's really brittle, as if I've bought more than one domain, the first one in the list could easily be the wrong one. I think we either need to fix the language here to drop the "new" in "Make your **new** domain your primary address", or update the logic to actually find the newly purchased domain.

@fditrapani, do you have a preference on which way this ought to go?

Reply:
Honestly I'd rather have the "new" part dropped, though "new" would apply to any domain the user would purchase during the month of this test.